model/src/cfd.rs

```diff
@@ -1403,10 +1379,45 @@ impl Cfd {
         }))
     }
 
+    fn event_with_reason(&self, event: EventKind, reason: anyhow::Error) -> CfdEvent {
```

Comment:
I find the name `reason` weird. Can we call it `error` ? Same for `log_cfd_event`

Reply:
We can - we used `reason` in rejection cases and `error` in failed cases. I generally don't like the `reason` terminology. we can opt to rename this throughout, but have to be careful not to break database backwards compatibility (I don't think we save the reason, so it should be fine, but this should be double-checked).

Reply:
throwing bunch of other names that could work:
- `detail` 
- `context` (a bit overloaded in the codebase)

Note: I would be hesitant to remove "reason" wholesale, I think it has its merit in other places - e.g. as "reason for rejecting an offer / settlement / something else" it is perfectly valid. I agree with @bonomat that in this context it is a bit awkward, because we abstracted away from these particular events here.